parlai/core/dict.py

```diff
@@ -383,46 +377,45 @@ def tokenize(self, text, building=False):
 
     def bpe_tokenize(self, text):
         """Returns a sequence of BPE-tokens from the text."""
-        if self.lower:
-            text = text.lower()
         return self.bpehelper.tokenize(text)
 
-    def finalize(self):
-        """
-        Finalize and freeze the dictionary.
-
-        If using BPE tokenization, performs the codec learning.
-        """
-
-        if self.tokenizer != 'bpe':
-            # only BPE needs the second pass
-            return
-
-        # Let the BPE model learn its codecs, then use the encodings to
-        # populate the DictionaryAgent. This could be moved to inside the
-        # _BPEHelper, but would require careful accounting to ensure vocabulary
-        # counts are correct.
-        if self.bpehelper.finalize():
-            for line in self.bpehelper.training_data:
-                self.add_to_dict(self.bpehelper.tokenize(line))
-
     def add_to_dict(self, tokens):
         """ Builds dictionary from the list of provided tokens."""
+        self.built = False
         for token in tokens:
             self.add_token(token)
             self.freq[token] += 1
 
     def remove_tail(self, min_freq):
+        """Removes elements below the frequency cutoff from the dictionary."""
         to_remove = []
         for token, freq in self.freq.items():
             if freq < min_freq:
                 # queue up removals since can't mutate dict during iteration
                 to_remove.append(token)
-                # other dicts can be modified as we go
-                idx = self.tok2ind.pop(token)
-                del self.ind2tok[idx]
+
         for token in to_remove:
             del self.freq[token]
+            idx = self.tok2ind.pop(token)
+            del self.ind2tok[idx]
+
+    def remove_non_bpe(self):
```

Comment:
should this be `_remove_non_bpe`? It seems like a good idea to signal to users that this isn't something they should call themselves.

Reply:
agreed, good call